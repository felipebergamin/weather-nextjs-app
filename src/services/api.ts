import axios from 'axios';

export default axios.create({
  baseURL: process.env.NEXT_PUBLIC_API_URL,
  params: {
    appid: {
      appid: process.env.API_KEY,
    },
  },
});
