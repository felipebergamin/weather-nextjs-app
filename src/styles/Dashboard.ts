import styled, { css } from 'styled-components';
import { ButtonBack, ButtonNext } from 'pure-react-carousel';

import { Card } from '~/components';

export const HorizontalCard = styled(Card)`
  display: flex;
`;

export const CenterContent = styled.div`
  display: flex;
  align-self: 'center';
  flex: 1;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const BaseCarouselButton = css`
  background: transparent;
  border: none;
`;

export const CarouselButtonNext = styled(ButtonNext)`
  ${BaseCarouselButton}
`;

export const CarouselButtonBack = styled(ButtonBack)`
  ${BaseCarouselButton}
`;
