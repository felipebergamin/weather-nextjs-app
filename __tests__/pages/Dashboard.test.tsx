import { render } from '@testing-library/react';
import AxiosMock from 'axios-mock-adapter';
import { address } from 'faker';

import Dashboard, { getServerSideProps } from '~/pages/dashboard/[lat]/[lon]';
import api from '~/services/api';

const mockAdapter = new AxiosMock(api);

describe('Dashboard screen', () => {
  it('rendered component should match snapshot', () => {
    const { baseElement } = render(
      <Dashboard
        current={{
          name: 'City Name',
          sys: {
            country: 'COUNTRY',
          },
          weather: [
            {
              id: 803,
              main: 'Clouds',
              description: 'broken clouds',
              icon: '04n',
            },
          ],
          main: {
            temp: 69.8,
            feels_like: 64.26,
            temp_min: 69.8,
            temp_max: 69.8,
            pressure: 1013,
            humidity: 36,
            sea_level: 1013,
            grnd_level: 934,
          },
        }}
        forecast={{
          list: [
            {
              dt: 1602374400,
              main: {
                temp: 71.42,
                feels_like: 68.68,
                temp_min: 68.68,
                temp_max: 71.42,
                pressure: 1011,
                sea_level: 1011,
                grnd_level: 933,
                humidity: 53,
                temp_kf: 1.52,
              },
              weather: [
                {
                  id: 803,
                  main: 'Clouds',
                  description: 'broken clouds',
                  icon: '04n',
                },
              ],
              clouds: {
                all: 78,
              },
              wind: {
                speed: 6.73,
                deg: 140,
              },
              visibility: 10000,
              pop: 0,
              sys: {
                pod: 'n',
              },
              dt_txt: '2020-10-11 00:00:00',
            },
            {
              dt: 1602385200,
              main: {
                temp: 66.54,
                feels_like: 61.32,
                temp_min: 64.47,
                temp_max: 66.54,
                pressure: 1013,
                sea_level: 1013,
                grnd_level: 934,
                humidity: 60,
                temp_kf: 1.15,
              },
              weather: [
                {
                  id: 801,
                  main: 'Clouds',
                  description: 'few clouds',
                  icon: '02n',
                },
              ],
              clouds: {
                all: 22,
              },
              wind: {
                speed: 10.51,
                deg: 139,
              },
              visibility: 10000,
              pop: 0,
              sys: {
                pod: 'n',
              },
              dt_txt: '2020-10-11 03:00:00',
            },
          ],
        }}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('getServerSideProps should fetch data', async () => {
    mockAdapter.onGet().reply(200, {});
    const [lat, lon] = [address.latitude(), address.longitude()];

    const response = await getServerSideProps({
      query: {
        lat,
        lon,
      },
      resolvedUrl: '',
      req: null,
      res: null,
    });

    expect(
      mockAdapter.history.get.map(({ url, params }) => ({
        url,
        params,
      })),
    ).toMatchObject([
      {
        url: 'weather',
        params: {
          lat,
          lon,
        },
      },
      {
        url: 'forecast',
        params: {
          lat,
          lon,
        },
      },
    ]);
    expect(response).toMatchSnapshot({
      props: {
        current: {},
        forecast: {},
      },
    });
  });
});
