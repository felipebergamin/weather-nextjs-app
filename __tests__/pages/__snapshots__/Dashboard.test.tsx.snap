// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard screen getServerSideProps should fetch data 1`] = `
Object {
  "props": Object {
    "current": Object {},
    "forecast": Object {},
  },
}
`;

exports[`Dashboard screen rendered component should match snapshot 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      title="Weather now"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          FoWeather
        </h6>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
      style="padding-top: 1.5rem; height: 100%;"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        Current weather at 
        City Name
        , 
        COUNTRY
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <img
            alt="ilustrating weather state"
            draggable="false"
            src="undefined/04n@4x.png"
          />
          <div>
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              Clouds
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              broken clouds
            </p>
            <span
              class="MuiTypography-root MuiTypography-caption"
            >
              Humidity 
              36
              %
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            69.8
             ℉
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Min: 
            69.8
             ℉ / Max: 
            69.8
             ℉
          </p>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            Feels like 
            64.26
             ℉
          </span>
        </div>
      </div>
      <div
        class="carousel"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-1"
          >
            <button
              aria-label="previous"
              class="buttonBack___1mlaL carousel__back-button sc-hKFyIo lcbVaY"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
          >
            <div
              aria-live="polite"
              class="horizontalSlider___281Ls carousel__slider carousel__slider--horizontal"
              role="listbox"
              tabindex="0"
            >
              <div
                class="carousel__slider-tray-wrapper carousel__slider-tray-wrap--horizontal"
              >
                <ul
                  class="sliderTray___-vHFQ sliderAnimation___300FY carousel__slider-tray carousel__slider-tray--horizontal"
                  style="width: 66.66666666666667%; transform: translateX(0%) translateX(0px); flex-direction: row;"
                >
                  <li
                    aria-selected="true"
                    class="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
                    role="option"
                    style="width: 50%; padding-bottom: 60%;"
                    tabindex="0"
                  >
                    <div
                      class="slideInner___2mfX9 carousel__inner-slide"
                    >
                      <div
                        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
                        style="margin: 0.25rem;"
                      >
                        <div
                          class="MuiCardContent-root"
                          style="display: flex; flex-direction: column;"
                        >
                          <div
                            class="MuiCardMedia-root"
                            style="background-image: url(undefined/04n@4x.png); width: 100%; height: 120px; background-size: contain;"
                          />
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            <strong>
                              today at 9:00 PM
                            </strong>
                          </p>
                          <span
                            class="MuiTypography-root MuiTypography-caption"
                          >
                            Temperature: 
                            71.42
                             ℉
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption"
                          >
                            Feels like: 
                            68.68
                             ℉
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption"
                          >
                            Humidity: 
                            71.42
                            %
                          </span>
                        </div>
                      </div>
                    </div>
                  </li>
                  <li
                    aria-selected="true"
                    class="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
                    role="option"
                    style="width: 50%; padding-bottom: 60%;"
                    tabindex="0"
                  >
                    <div
                      class="slideInner___2mfX9 carousel__inner-slide"
                    >
                      <div
                        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
                        style="margin: 0.25rem;"
                      >
                        <div
                          class="MuiCardContent-root"
                          style="display: flex; flex-direction: column;"
                        >
                          <div
                            class="MuiCardMedia-root"
                            style="background-image: url(undefined/02n@4x.png); width: 100%; height: 120px; background-size: contain;"
                          />
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            <strong>
                              tomorrow at 12:00 AM
                            </strong>
                          </p>
                          <span
                            class="MuiTypography-root MuiTypography-caption"
                          >
                            Temperature: 
                            66.54
                             ℉
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption"
                          >
                            Feels like: 
                            61.32
                             ℉
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption"
                          >
                            Humidity: 
                            66.54
                            %
                          </span>
                        </div>
                      </div>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-1"
          >
            <button
              aria-label="next"
              class="buttonNext___2mOCa carousel__next-button sc-dlnjPT eZjeJu"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
